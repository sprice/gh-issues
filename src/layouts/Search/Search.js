import React from "react"
import { Consumer } from "../../SearchContext"

class Search extends React.Component {
  render() {
    return (
      <Consumer>
        {context => (
          <div className="search-page">
            <h1>GitHub Issue Viewer</h1>
            <label className="search-label" htmlFor="search">
              <input
                type="text"
                id="search"
                className="search-box"
                placeholder="Type a link to a GitHub repo!"
                value={context.githubUrl}
                onChange={context.handleUrlChange}
              />
            </label>
            <button onClick={context.searchIssues}>View Issues</button>
            {context.error ? (
              <h2 className="error">{context.error.message}</h2>
            ) : null}
          </div>
        )}
      </Consumer>
    )
  }
}

export default Search
