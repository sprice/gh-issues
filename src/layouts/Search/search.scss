.search-page {
  background-color: $pink;
  height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.search-page h1 {
  margin-bottom: 1rem;
  font-size: 1.25rem;
}

.search-box {
  padding: 0.75rem 0.75rem 0.75rem 2rem;
  width: 12rem;
  font-size: 1rem;
}

.search-box::placeholder {
  color: $light-grey;
  font-size: 1rem;
}

.search-label {
  position: relative;
}

.search-label:before {
  content: "";
  position: absolute;
  left: 5px;
  top: 15px;
  bottom: 0;
  width: 20px;

  background-color: $light-grey;
  mask-image: url("../icons/search.svg");
  mask-repeat: no-repeat;
  -webkit-mask-image: url("../icons/search.svg");
  -webkit-mask-repeat: no-repeat;
}

button {
  margin-top: 1rem;
  padding: 1rem;
  background-color: $white;
  border: none;
  border-radius: 10px;
  color: $pink;
  font-weight: 700;
  font-size: 1.75rem;
  box-shadow: 3px 3px 10px 0px rgba($light-grey, 0.75);
}

button:hover {
  cursor: pointer;
}

button:active {
  transform: translateY(1px);
  box-shadow: 3px 2px 10px 0px rgba($light-grey, 0.75);
}

.error {
  margin-top: 3rem;
}

@media (min-width: $breakpoint-small) {
  .search-page h1 {
    font-size: 3rem;
  }

  .search-box {
    width: 30rem;
    padding: 0.75rem 0.75rem 0.75rem 3rem;
  }

  .search-box::placeholder {
    position: relative;
    font-size: 1.75rem;
    top: 5px; /* Chrome is weird */
  }

  .search-label:before {
    left: 10px;
    top: 10px;
    width: 30px;
  }
}

@media (min-width: $breakpoint-medium) {
  .search-box {
    width: 40rem;
  }
}
