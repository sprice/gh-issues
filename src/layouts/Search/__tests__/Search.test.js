import React from "react"
import { create } from "react-test-renderer"
import Search from "../Search"

test("Snapshop of Search", () => {
  const c = create(<Search />)
  expect(c.toJSON()).toMatchSnapshot()
})
