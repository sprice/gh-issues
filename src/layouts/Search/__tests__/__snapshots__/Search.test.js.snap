// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshop of Search 1`] = `
<div
  className="search-page"
>
  <h1>
    GitHub Issue Viewer
  </h1>
  <label
    className="search-label"
    htmlFor="search"
  >
    <input
      className="search-box"
      id="search"
      onChange={[Function]}
      placeholder="Type a link to a GitHub repo!"
      type="text"
      value=""
    />
  </label>
  <button
    onClick={[Function]}
  >
    View Issues
  </button>
</div>
`;
